stateproof/signer.go

```diff
@@ -54,13 +55,32 @@ func (spw *Worker) signer(latest basics.Round) {
 			spw.invokeBuilder(nextRnd)
 			nextRnd++
 
+			nxtstateProof := hdr.StateProofTracking[protocol.StateProofBasic].StateProofNextRound
+			if nxtstateProof > prevStateProof {
+				spw.attemptKeyDeletionPriorToRound(prevStateProof)
```

Comment:
will it be simpler to move this code to the `builder` function next to where old signatures are being removed?